sample/redo/lib/main.dart

```diff
@@ -116,41 +116,42 @@ class _MyHomePageState extends State<MyHomePage> {
       name: loginResult.userId,
       clientViewAuth: loginResult.userId,
     );
-    _replicache.onChange = _load;
     _replicache.onPull = _handlePull;
     _replicache.getClientViewAuth = _getAuthToken;
 
     setState(() {
       _loginResult = loginResult;
     });
 
-    await _load();
-  }
-
-  Future<void> _load() async {
-    final res = await _replicache.query((tx) async {
-      return await Future.wait(
-          [tx.scan(prefix: '/list/'), tx.scan(prefix: prefix)]);
+    _listIdStream().listen((listIds) {
+      setState(() {
+        if ((_selectedListId == null || !listIds.contains(_selectedListId)) &&
+            listIds.isNotEmpty) {
+          _selectedListId = listIds.first;
+        }
+        _listIds = listIds;
+      });
     });
 
-    final listIdScanItems = res[0];
-    final todosScanItems = res[1];
-
-    List<int> listIds = List.from(listIdScanItems
-        .map((item) => int.parse(item.key.substring('/list/'.length))));
-    List<Todo> allTodos = List.from(todosScanItems
-        .map((item) => Todo.fromJson(stripPrefix(item.key), item.value)));
-
-    setState(() {
-      if ((_selectedListId == null || !listIds.contains(_selectedListId)) &&
-          listIds.length > 0) {
-        _selectedListId = listIds[0];
-      }
-      _listIds = listIds;
-      _allTodos = allTodos;
+    _todoStream().listen((allTodos) {
+      setState(() {
+        _allTodos = allTodos;
+      });
     });
   }
 
+  Stream<Iterable<int>> _listIdStream() => _replicache.subscribe(
```

Comment:
I played around with a lot of different ways to structure this and this was the cleanest thing I could come up with.

Reply:
It looks like roughly what I had in mind. How does it feel to use?

Reply:
Any ideas for improvements?

Reply:
One can make it a bit more verbose to make it less cryptic but it this is the best I got.